src/api/app/controllers/concerns/webui/notifications_filter.rb

```diff
@@ -1,33 +1,56 @@
 module Webui::NotificationsFilter
   extend ActiveSupport::Concern
 
-  # It's just a case...
+  # It's a sequence of multiple conditions combinations
   # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
   def filter_notifications_by_type(notifications, filter_type)
-    case filter_type
-    when 'comments'
-      notifications.for_comments
-    when 'requests'
-      notifications.for_requests
-    when 'incoming_requests'
-      notifications.for_incoming_requests(User.session!)
-    when 'outgoing_requests'
-      notifications.for_outgoing_requests(User.session!)
-    when 'relationships_created'
-      notifications.for_relationships_created
-    when 'relationships_deleted'
-      notifications.for_relationships_deleted
-    when 'build_failures'
-      notifications.for_failed_builds
-    when 'reports'
-      notifications.for_reports
-    when 'workflow_runs'
-      notifications.for_workflow_runs
-    when 'appealed_decisions'
-      notifications.for_appealed_decisions
-    else # all
-      notifications
-    end
+    return notifications if filter_type == ['all']
+
+    relations_type = []
+    relations_type << notifications.for_comments if filter_type.include?('comments')
```

Comment:
I think the `if filter_type.include?('comments')` is not really needed. If we select the comments filter, but there are none, it still works...

Reply:
Here we are deciding if we need to add `notifications.for_comments` depending on whether the `comments` filter is selected or not. If the `filter_type` does not `include` the `comments` filter field it means we don't want the take the `notifications.for_comments` subset and add it to the resulting query.

Reply:
Ah right, NVM